### ISSUE TITLE
syscall() proto not found on illumos

### ISSUE THREAD

**Issue by mtelka** [Perl/perl5#21042]
**Description**
The Perl's `Configure` script is trying to find the `syscall` proto in `unistd.h` file:
```
: see if prototype for syscall is available
echo " "
set d_syscallproto syscall $i_unistd unistd.h
eval $hasproto

```
This works correctly on Linux where the `syscall` prototype is in `unistd.h`, but it does not work on illumos (and maybe on Solaris too) where the `syscall` prototype is in the `sys/syscall.h` file.  As a result the `HAS_SYSCALL_PROTO` is not detected properly and when there is some 3rd party software built it could fail, like this (the output below is from `BSD::Resource` CPAN module compilation):
```
Running Mkbootstrap for Resource ()
chmod 644 "Resource.bs"
"/usr/perl5/5.36/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Resource.bs blib/arch/auto/BSD/Resource/Resource.bs 644
"/usr/perl5/5.36/bin/perl" "/usr/perl5/5.36/lib/ExtUtils/xsubpp"  -typemap '/usr/perl5/5.36/lib/ExtUtils/typemap'  Resource.xs > Resource.xsc
mv Resource.xsc Resource.c
/usr/gcc/10/bin/gcc -m64 -c   -D_REENTRANT -m64 -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE64_SOURCE -D_FORTIFY_SOURCE=2 -DPERL_USE_SAFE_PUTENV -O3   -DVERSION=\"1.2911\" -DXS_VERSION=\"1.2911\" -fPIC "-I/usr/perl5/5.36/lib/i86pc-solaris-thread-multi-64/CORE"  -DRlim_t=rlim_t Resource.c
In file included from /usr/include/sys/procfs.h:63,
                 from Resource.xs:57:
/usr/include/sys/syscall.h:517:13: error: conflicting types for 'syscall'
  517 | extern long syscall(int, ...);
      |             ^~~~~~~
In file included from Resource.xs:14:
/usr/perl5/5.36/lib/i86pc-solaris-thread-multi-64/CORE/perl.h:1152:14: note: previous declaration of 'syscall' was here
 1080 | EXTERN_C int syscall(int, ...);
      |              ^~~~~~~
make[1]: *** [Makefile:333: Resource.o] Error 1
```

**Steps to Reproduce**
Compile Perl on OpenIndiana (an illumos distribution).

```
$ grep syscall build.log
syscall() found.
syscall() prototype NOT found.
Have syscall()... looking for syscall.h... found <sys/syscall.h>.
$
```

**Expected behavior**

The `syscall` prototype is found on illumos.

**Perl configuration**
```
# perl -V
Summary of my perl5 (revision 5 version 36 subversion 0) configuration:
   
  Platform:
    osname=solaris
    osvers=2.11
    archname=i86pc-solaris-thread-multi-64
    uname='sunos hipster.openindiana.org 5.11 illumos-b172429eb7 i86pc i386 i86pc '
    config_args='-de -Dmksymlinks -Ulocincpth= -Uloclibpth= -Dprefix=/usr/perl5/5.36 -Dprivlib=/usr/perl5/5.36/lib -Dsitelib=/usr/perl5/site_perl/5.36 -Dvendorprefix=/usr/perl5/5.36 -Dvendorlib=/usr/perl5/vendor_perl/5.36 -Dman3dir=/usr/perl5/5.36/man/man3perl -Dman3ext=3perl -Dmyhostname=localhost -Dcf_email=oi-dev@openindiana.org -Dcf_by=perl-bugs -Duseshrplib -Dlibperl=libperl.so -Dusedtrace -Dusethreads -Dcc=/usr/gcc/10/bin/gcc -m64 -Doptimize=-O3 -Dlibpth=/lib/64 /usr/lib/64'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='/usr/gcc/10/bin/gcc -m64'
    ccflags ='-D_REENTRANT -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -m64 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -DPERL_USE_SAFE_PUTENV'
    optimize='-O3'
    cppflags='-D_REENTRANT -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong'
    ccversion=''
    gccversion='10.4.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='/usr/gcc/10/bin/gcc -m64'
    ldflags =' -fstack-protector-strong -m64 '
    libpth=/lib/64 /usr/lib/64 /usr/gcc/10/lib /usr/lib /usr/ccs/lib
    libs=-lpthread -lsocket -lnsl -lgdbm -ldb -ldl -lm -lc
    perllibs=-lpthread -lsocket -lnsl -ldl -lm -lc
    libc=/lib/libc.so
    so=so
    useshrplib=true
    libperl=libperl.so
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='  -R /usr/perl5/5.36/lib/i86pc-solaris-thread-multi-64/CORE'
    cccdlflags='-fPIC'
    lddlflags=' -shared -m64 -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    PERL_USE_SAFE_PUTENV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_REENTRANT_API
    USE_THREAD_SAFE_LOCALE
  Built under solaris
  Compiled at Feb 19 2023 16:34:03
  @INC:
    /usr/perl5/site_perl/5.36/i86pc-solaris-thread-multi-64
    /usr/perl5/site_perl/5.36
    /usr/perl5/vendor_perl/5.36/i86pc-solaris-thread-multi-64
    /usr/perl5/vendor_perl/5.36
    /usr/perl5/5.36/lib/i86pc-solaris-thread-multi-64
    /usr/perl5/5.36/lib
```

**Comment by tonycoz** [comment]
This should be fixed in blead by the combination of 615e36588766a246d9d5f1bbd6065cc888a26df6, 56713ab45217704e3741b391feb10b4ab37c188b and d45c0d11c1c4f0389f94233f01562341113ed498

**Comment by tonycoz** [comment]
Closing
